stdlib/domain.ml

```diff
@@ -172,17 +92,33 @@ let do_before_first_spawn () =
     first_spawn_function := (fun () -> ())
   end
 
-let at_exit_key = DLS.new_key (fun () -> (fun () -> ()))
+module IMap = Map.Make (struct type t = int let compare = (-) end)
```

Comment:
I would use `Int.compare` instead of `(-)`, because I am not sure that `(-)` is correct in overflow situation. This does not matter in this specific case as we will only see small non-negative integers, but still `let compare = (-)` is a code smell to me.

Reply:
Fixed.